src/makeCalculator.js

```diff
@@ -4,7 +4,36 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+
+    add(num) {
+      calculator.result += num;
+    },
+    subtract(num) {
+      calculator.result -= num;
+    },
+    multiply(num) {
+      calculator.result *= num;
+    },
+    divide(num) {
+      calculator.result /= num;
+    },
+
+    reset() {
+      calculator.result = 0;
+
+      return calculator; // Allow chaining
+    },
+
+    operate(callback, num) {
```

Comment:
[CODE KNOWLEDGE] - It's a good practice to check if the provided callback is indeed a function to avoid runtime errors. Consider adding a check like 'if (typeof callback === 'function') {...}' before invoking the callback.